pkg/controllers/deprovisioning/types.go

```diff
@@ -61,6 +63,10 @@ func NewCandidate(ctx context.Context, kubeClient client.Client, clk clock.Clock
 		v1alpha5.ProvisionerNameLabelKey,
 	} {
 		if _, ok := node.Labels()[label]; !ok {
+			// This means that we don't own the candidate which means we shouldn't fire an event for it
+			if label != v1alpha5.ProvisionerNameLabelKey {
+				recorder.Publish(deprovisioningevents.Blocked(node.Node, node.Machine, fmt.Sprintf("required label %q doesn't exist", label))...)
```

Comment:
Should we modify the recorder to log any `deprovisioningevents.Blocked` events for now?

Reply:
SGTM. I could see aligning our events with our logger generally -- we may need to remove some redundant log lines if we do this, though. We'd likely want to do this at DEBUG.

Reply:
Yeah, I was just discussing with @tzneal that this seemed like it would log quite a bit considering this is a filtering mechanism that would potentially remove inflight nodes or nodes that haven't yet initialized yet. Also, each deprovsioner calls this function so we'd have to dedupe across provisioners.